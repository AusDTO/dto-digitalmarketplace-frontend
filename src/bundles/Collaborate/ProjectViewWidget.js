import React from 'react'
import { Provider } from 'react-redux'
import RegisterComponent from '../../RegisterComponent'

import createStore from './redux/create'
import ProjectView from './components/ProjectView'

export const ProjectViewWidget = (props) => {
  const store = createStore(props)
  return (
    <Provider store={store}>
      <ProjectView />
    </Provider>
  )
}

export default new RegisterComponent({ 'project-view': ProjectViewWidget })
