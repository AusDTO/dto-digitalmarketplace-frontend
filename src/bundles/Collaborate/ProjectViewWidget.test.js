// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');

import React from 'react';
import { ProjectViewWidget } from './ProjectViewWidget';
import renderer from 'react-test-renderer';

test('ProjectViewWidget renders without errrors', () => {
  const state = {
    casestudy: {
      title: 'A Case Study',
      opportunity: 'A Case Study Opportunity',
      client: 'A Case study client',
      approach: 'A simple approach',
      timeframe: '2016',
      outcome: ['a', 'b'],
      projectLinks: ['http://facebook.com']
    }
  }
  const component = renderer.create(
    <ProjectViewWidget {...state} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})