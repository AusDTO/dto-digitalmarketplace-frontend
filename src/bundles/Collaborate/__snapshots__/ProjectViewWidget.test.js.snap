// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectViewWidget renders without errrors 1`] = `
<div>
  <section
    styleName="full-header"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12"
      >
        <h1
          tabIndex="-1"
        >
          A Case Study
        </h1>
      </div>
    </div>
    <div
      className="row"
    >
      <article
        className="col-xs-12 col-md-8 col-sm-7"
      >
        <p
          styleName="summary"
        />
        <p>
          <strong>
            Project stage
          </strong>
           
        </p>
      </article>
      <article
        className="col-xs-12 col-sm-8 col-sm-push-1 col-md-3 col-md-push-1"
      >
        <div
          style={
            Object {
              "backgroundColor": "white",
            }
          }
          styleName="tile"
        >
          <span>
            Council
          </span>
          <b>
            A Case study client
          </b>
          <a
            href="mailto:undefined"
            role="button"
          >
            Email contact
          </a>
        </div>
      </article>
    </div>
  </section>
  <section
    id="project__ProjectView"
    styleName="center"
  >
    <div
      className="row"
    >
      <aside
        className="col-sm-3 col-xs-12"
      />
      <article
        className="col-sm-7 col-xs-12"
        role="main"
      >
        <section>
          <h2>
            What's the key problem to solve?
          </h2>
          <p
            className="freetext"
          >
            <span>
              A simple problem
              <br />
            </span>
          </p>
        </section>
        <section>
          <h2>
            Outcomes
          </h2>
          <ul>
            <li>
              a
            </li>
            <li>
              b
            </li>
          </ul>
        </section>
        <div
          className="project__actions"
        />
      </article>
    </div>
  </section>
</div>
`;
